sdk/python/feast/infra/redis_provider.py

```diff
@@ -0,0 +1,276 @@
+import json
+import os
+import struct
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+import mmh3
+import pandas as pd
+from google.protobuf.timestamp_pb2 import Timestamp
+from redis import Redis
+from rediscluster import RedisCluster
+from tqdm import tqdm
+
+from feast import FeatureTable, utils
+from feast.entity import Entity
+from feast.feature_view import FeatureView
+from feast.infra.offline_stores.helpers import get_offline_store_from_sources
+from feast.infra.provider import (
+    Provider,
+    RetrievalJob,
+    _convert_arrow_to_proto,
+    _get_column_names,
+    _run_field_mapping,
+)
+from feast.protos.feast.storage.Redis_pb2 import RedisKeyV2 as RedisKeyProto
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.registry import Registry
+from feast.repo_config import RedisOnlineStoreConfig, RepoConfig
+
+EX_SECONDS = 253402300799
+
+
+class RedisProvider(Provider):
+    _db_path: Path
+
+    def __init__(self, config: RepoConfig):
+        assert isinstance(config.online_store, RedisOnlineStoreConfig)
+
+    def update_infra(
+        self,
+        project: str,
+        tables_to_delete: Sequence[Union[FeatureTable, FeatureView]],
+        tables_to_keep: Sequence[Union[FeatureTable, FeatureView]],
+        entities_to_delete: Sequence[Entity],
+        entities_to_keep: Sequence[Entity],
+        partial: bool,
+    ):
+        pass
+
+    def teardown_infra(
+        self,
+        project: str,
+        tables: Sequence[Union[FeatureTable, FeatureView]],
+        entities: Sequence[Entity],
+    ) -> None:
+        # according to the repos_operations.py we can delete the whole project
+        client = self._get_client()
+
+        tables_join_keys = [[e for e in t.entities] for t in tables]
+        for table_join_keys in tables_join_keys:
+            redis_key_bin = _redis_key(
+                project, EntityKeyProto(join_keys=table_join_keys)
+            )
+            keys = client.keys(f"{redis_key_bin}*")
```

Comment:
This is a general Redis tip, but you should never use the `KEYS` command in production because it loads all matching keys into memory. That can block the server and use too much memory. Instead, use the `SCAN` family of commands. In this case, you can use redis-py's `scan_iter()` method -- something like:

```
keys = [k for k in redis.scan_iter(match=f'{redis_key_bin}*', count=100)]
if keys:
    client.unlink(*keys)
```

Reply:
Thanks @abrookins, good catch!

Reply:
Thanks @abrooking I have fixed this :)

Reply:
Awesome! 👍